platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/renderer/glsurfaceview/GLSurfaceViewMapRenderer.java

```diff
@@ -39,13 +38,17 @@ public GLSurfaceViewMapRenderer(Context context,
       @Override
       public void surfaceCreated(SurfaceHolder holder) {
         super.surfaceCreated(holder);
-        hasSurface.set(true);
+        synchronized (lock) {
+          hasSurface = true;
+        }
       }
 
       @Override
       public void surfaceDestroyed(SurfaceHolder holder) {
         super.surfaceDestroyed(holder);
-        hasSurface.set(false);
+        synchronized (lock) {
+          hasSurface = false;
+        }
         nativeReset();
```

Comment:
this should also be synchronized as that's the actual critical part that manipulates the pointer